src/nvim/move.c

```diff
@@ -1504,10 +1508,8 @@ bool scrollup(win_T *wp, linenr_T line_count, bool byfold)
 
   check_topfill(wp, false);
 
-  if (hasAnyFolding(wp)) {
-    // Make sure w_topline is at the first of a sequence of folded lines.
-    hasFolding(wp, wp->w_topline, &wp->w_topline, NULL);
-  }
+  // Make sure w_topline is at the first of a sequence of folded lines.
+  hasFolding(wp, wp->w_topline, &wp->w_topline, NULL);
```

Comment:
This change is a bit strange. Concealed lines don't affect `hasFolding()`, so why doesn't `hasFolding()` now have to be called even if there are no folds?

Reply:
It's been a while but I think I just removed it to match other callsites in this file. It's usually called unconditionally (it has this condition as an early return).